src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2798,12 +2804,17 @@ protected override void RemoveItem(string path, bool recurse)
                     return;
                 }
 
+#if !UNIX
```

Comment:
As above, the code is easier to read if you use a #if UNIX and contain the UNIX lines completely.  Its a simple if/else with a single statement for UNIX but  looks unwieldy with the current #if/#else placement.